### PR TITLE
backport 2.7:Add tests for mbedtls_cipher_crypt API

### PR DESCRIPTION
**Pull request by RonEld**
## Description
Backport of #1099 to mbedtls-2.7

## Status
**READY**

### CODE REVIEW

File: library/cipher.c [hunk 1]
```diff
@@ -215,9 +215,13 @@ int mbedtls_cipher_set_iv( mbedtls_cipher_context_t *ctx,
                    const unsigned char *iv, size_t iv_len )
 {
     size_t actual_iv_size;
-
-    if( NULL == ctx || NULL == ctx->cipher_info || NULL == iv )
+    if( NULL == ctx || NULL == ctx->cipher_info )
         return( MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA );
+    else if( NULL == iv && iv_len != 0  )
```
Comment (andresag01) [inline library/cipher.c]: Minor style: Please remove double space before `)`

File: library/cipher.c [hunk 2]
```diff
@@ -215,9 +215,13 @@ int mbedtls_cipher_set_iv( mbedtls_cipher_context_t *ctx,
                    const unsigned char *iv, size_t iv_len )
 {
     size_t actual_iv_size;
-
-    if( NULL == ctx || NULL == ctx->cipher_info || NULL == iv )
+    if( NULL == ctx || NULL == ctx->cipher_info )
```
Comment (andresag01) [inline library/cipher.c]: Minor style: Please put back whitespace removed by this patch.

File: library/cipher.c [hunk 3]
```diff
@@ -233,9 +237,11 @@ int mbedtls_cipher_set_iv( mbedtls_cipher_context_t *ctx,
         if( actual_iv_size > iv_len )
             return( MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA );
     }
-
-    memcpy( ctx->iv, iv, actual_iv_size );
-    ctx->iv_size = actual_iv_size;
+    if ( actual_iv_size != 0 )
```
Comment (andresag01) [inline library/cipher.c]: Minor Style: Please put back empty line before these statements removed in this patch

File: tests/suites/test_suite_cipher.function [hunk 4]
```diff
@@ -690,6 +701,55 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_MODE_WITH_PADDING */
+void test_vec_crypt( int cipher_id, int operation, char *hex_key,
+                   char *hex_iv, char *hex_input, char *hex_result,
```
Comment (andresag01) [inline tests/suites/test_suite_cipher.function]: Minor style: Please align function arguments

**Comment by sbutcher-arm** [comment]
This is waiting on the original PR #1099 to be rebased before being merged, so it can be merged at the same time.

**Comment by sbutcher-arm** [comment]
The original PR this is a backport of, #1099, has conflicts and needs rebasing, therefore demoting, and removing the 'ready for merge' label.

**Comment by sbutcher-arm** [comment]
The original PR this is a backport of, #1099, still has conflicts and needs rebasing, therefore demoting, and removing the 'ready for merge' label.

**Pull request merged**